File: 승준/20230531/birth_5635.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.time.LocalDate;
import java.time.Period;
import java.util.StringTokenizer;
public class birth_5635 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st_n = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st_n.nextToken());

        String[][] userInfoLst = new String[n][4];
        for (int i = 0; i < n; i++) {
            StringTokenizer st_userInfo = new StringTokenizer(br.readLine());
            for (int j = 0; j < 4; j++) {
                if (st_userInfo.hasMoreTokens()) {
                    String token = st_userInfo.nextToken();
                    userInfoLst[i][j] = token;
                }
            }
        }

        int[] mx_age = {0, 0, 0};
        int[] mn_age = {1000, 1000, 1000};
        String oldest = "";
        String youngest = "";

        for (int i = 0; i < n; i++) {
            int birthYear = Integer.parseInt(userInfoLst[i][3]);
            int birthMonth = Integer.parseInt(userInfoLst[i][2]);
            int birthDay = Integer.parseInt(userInfoLst[i][1]);

            LocalDate currentDate = LocalDate.now();
            LocalDate birthDate = LocalDate.of(birthYear, birthMonth, birthDay);

            Period period = Period.between(birthDate, currentDate);
            int[] age = {period.getYears(), period.getMonths(), period.getDays()};

            if (isGreater(age, mx_age)) {
                mx_age = age;
                oldest = userInfoLst[i][0];
            }

            if (isSmaller(age, mn_age)) {
                mn_age = age;
                youngest = userInfoLst[i][0];
            }
        }

        System.out.println(youngest);
        System.out.println(oldest);
    }

    private static boolean isGreater(int[] age1, int[] age2) {
        if (age1[0] > age2[0]) {
            return true;
        } else if (age1[0] == age2[0] && age1[1] > age2[1]) {
            return true;
        } else return age1[0] == age2[0] && age1[1] == age2[1] && age1[2] > age2[2];
    }

    private static boolean isSmaller(int[] age1, int[] age2) {
        if (age1[0] < age2[0]) {
            return true;
        } else if (age1[0] == age2[0] && age1[1] < age2[1]) {
            return true;
        } else return age1[0] == age2[0] && age1[1] == age2[1] && age1[2] < age2[2];
    }
}

File: 승준/20230531/birth_5635.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let n = 0;
const userInfoLst = [];

rl.on('line', (line) => {
    if (n === 0) {
      n = parseInt(line);
    } else {
      const tokens = line.split(' ');
      userInfoLst.push(tokens);
  
      if (userInfoLst.length === n) {
        processInput(userInfoLst);
        rl.close();
      }
    }
  }
);

function processInput(userInfoLst) {
    const mx_age = [0, 0, 0];
    const mn_age = [1000, 1000, 1000];
    let oldest = "";
    let youngest = "";

    for (let i = 0; i < userInfoLst.length; i++) {
        const birthYear = parseInt(userInfoLst[i][3]);
        const birthMonth = parseInt(userInfoLst[i][2]);
        const birthDay = parseInt(userInfoLst[i][1]);

        const currentDate = new Date();
        const birthDate = new Date(birthYear, birthMonth - 1, birthDay);

        const age = calculateAge(birthDate, currentDate);

        if (isGreater(age, mx_age)) {
            mx_age[0] = age[0];
            mx_age[1] = age[1];
            mx_age[2] = age[2];
            oldest = userInfoLst[i][0];
        }

        if (isSmaller(age, mn_age)) {
            mn_age[0] = age[0];
            mn_age[1] = age[1];
            mn_age[2] = age[2];
            youngest = userInfoLst[i][0];
        }
    }

    console.log(youngest);
    console.log(oldest);
}

function calculateAge(birthDate, currentDate) {
    const yearsDiff = currentDate.getFullYear() - birthDate.getFullYear();
    const monthsDiff = currentDate.getMonth() - birthDate.getMonth();
    const daysDiff = currentDate.getDate() - birthDate.getDate();

    let years = yearsDiff;
    let months = monthsDiff;
    let days = daysDiff;

    if (monthsDiff < 0 || (monthsDiff === 0 && daysDiff < 0)) {
        years--;
        months += 12;
    }

    if (daysDiff < 0) {
        months--;
        const prevMonthDte = new Date(currentDate.getFullYear(), currentDate.getMonth() -1, 0);
        days += prevMonthDte.getDate();
    }

    return [years, months, days];
}

function isGreater(age1, age2) {
  if (age1[0] > age2[0]) {
    return true;
  } else if (age1[0] === age2[0] && age1[1] > age2[1]) {
    return true;
  } else return age1[0] === age2[0] && age1[1] === age2[1] && age1[2] > age2[2];
}

function isSmaller(age1, age2) {
  if (age1[0] < age2[0]) {
    return true;
  } else if (age1[0] === age2[0] && age1[1] < age2[1]) {
    return true;
  } else return age1[0] === age2[0] && age1[1] === age2[1] && age1[2] < age2[2];
}